packages/react/menu/src/Menu.tsx

```diff
@@ -644,15 +644,19 @@ const MenuItem = React.forwardRef((props, forwardedRef) => {
   const context = useMenuContext(ITEM_NAME);
   const contentContext = useMenuContentContext(ITEM_NAME);
   const composedRefs = useComposedRefs(forwardedRef, ref);
+  const isPointerDownRef = React.useRef(false);
 
   const handleSelect = () => {
     const menuItem = ref.current;
     if (!disabled && menuItem) {
       const itemSelectEvent = new Event(ITEM_SELECT, { bubbles: true, cancelable: true });
       menuItem.addEventListener(ITEM_SELECT, (event) => onSelect?.(event), { once: true });
       menuItem.dispatchEvent(itemSelectEvent);
-      if (itemSelectEvent.defaultPrevented) return;
-      context.onRootClose();
+      if (itemSelectEvent.defaultPrevented) {
+        isPointerDownRef.current = false;
```

Comment:
Just to make sure I understand fully, we're resetting here because the menu won't close, the item won't unmount and so the ref won't be reset automatically on remount?

Reply:
the ref will always be reset automatically on remount, this is specifically for a case where someone prevents it from closing. Without this, this would break:

1. on the same item `pointer down -> pointer up`
2. `event.preventDefault()` in `onSelect`
3. on a different item `pointer down -> pointer move -> pointer up on the item from step 1`

That pointer up on the item from step one would think they pointered down on it too because it wouldn't have been reset when it was kept open.